### PR TITLE
[ready] Fix/roundabout segregated

### PR DESCRIPTION
**Pull request by MoKob**
Technically more an data-issue than anything else, but currently we could issue instructions against the turning direction of a roundabout for segregated roads merging into a roundabout at a single point.

This PR addresses the issue.

 - [ ] requires https://github.com/Project-OSRM/osrm-backend/pull/2248
 - [x] review done

### CODE REVIEW

File: src/engine/guidance/post_processing.cpp [hunk 1]
```diff
@@ -25,6 +25,28 @@ namespace engine
...
+        const auto type = static_cast<int>(step.maneuver.instruction.type);
+        const auto modifier = static_cast<int>(step.maneuver.instruction.direction_modifier);
+
+        std::cout << "\t[" << ++segment << "]: " << type << " " << modifier
+                  << " Duration: " << step.duration << " Distance: " << step.distance
+                  << " Geometry: " << step.geometry_begin << " " << step.geometry_end
+                  << " exit: " << step.maneuver.exit
+                  << " Intersections: " << step.maneuver.intersections.size() << " [";
+
+        for (auto intersection : step.maneuver.intersections)
```
Comment (daniel-j-h) [inline src/engine/guidance/post_processing.cpp]: `const auto intersection&`

File: src/extractor/guidance/intersection_generator.cpp [hunk 2]
```diff
@@ -207,6 +208,16 @@ Intersection IntersectionGenerator::mergeSegregatedRoads(Intersection intersecti
     if (intersection.size() == 1)
         return intersection;
 
+    bool is_connected_to_roundabout = false;
```
Comment (daniel-j-h) [inline src/extractor/guidance/intersection_generator.cpp]: can be const and initialized via a lambda?

File: src/extractor/guidance/roundabout_handler.cpp [hunk 3]
```diff
@@ -79,6 +80,43 @@ detail::RoundaboutFlags RoundaboutHandler::getRoundaboutFlags(
...
+        // only check actual outgoing edges
+        if (edge_data.reversed)
+        {
+
+            // remember whether we have seen the roundabout in-part
+            if (edge_data.roundabout)
+                past_roundabout_angle = true;
+
+            continue;
+        }
```
Comment (daniel-j-h) [inline src/extractor/guidance/roundabout_handler.cpp]: Maybe flatten to

    if reversed && roundabout

Comment (MoKob) [inline src/extractor/guidance/roundabout_handler.cpp]: Wouldn't be the same, would  it? I continue on all reverse edges, but only set the flag if its also a roundabout.

**Comment by TheMarex** [comment]
Please reopen the PR for `master`, `develop` is going away.

**Pull request closed without merging**